Kinvey/Kinvey/CustomEndpoint.swift

```diff
@@ -36,13 +42,28 @@ open class CustomEndpoint: NSObject {
         return request
     }
     
+    /// Executes a custom endpoint by name and passing the expected parameters.
+    @discardableResult
+    open static func executeArray(_ name: String, params: JsonDictionary? = nil, client: Client = sharedClient, completionHandler: ArrayCompletionHandler? = nil) -> Request {
```

Comment:
@heyzooi who determines whether the request is executed as `execute()` or `executeArray()`? From your changes, looks like its the developer's responsibility to call the correct API. Instead, can we overload a single `execute()` method that uses the type of the `completionHandler` to determine what to execute? 

Can